src/plugin.js

```diff
@@ -174,6 +174,12 @@ const contribAdsPlugin = function(options) {
 
     // TODO reset state to content-set here instead of in every contentupdate case
     reset() {
+      if (player.ads.snapshot.trackChangeHandler) {
```

Comment:
This listener is also removed when `restorePlayerSnapshot` is called. I added it here so it will also be cleaned up on contentupdate events per @ldayananda's request.